contractcourt/channel_arbitrator.go

```diff
@@ -507,6 +507,27 @@ func (c *ChannelArbitrator) stateStep(triggerHeight uint32,
 		log.Infof("ChannelArbitrator(%v): force closing "+
 			"chan", c.cfg.ChanPoint)
 
+		// Under normal operation, we can only enter
+		// StateBroadcastCommit via a user or chain trigger. On restart,
+		// this state may be reexecuted after closing the channel, but
+		// failing to commit to StateContractClosed. In that case, one
+		// of the three three close triggers will be presented,
+		// signifying that we should skip rebroadcasting, and straight
+		// to resolving the on-chain contract.
+		switch {
+		case coopCloseTrigger:
+			fallthrough
+		case localCloseTrigger:
+			fallthrough
+		case remoteCloseTrigger:
+			nextState = StateContractClosed
```

Comment:
No need for this assignment, `StateContractClosed` can be logged and returned instead.

Reply:
I think it is better to assign then `break` really, as this is what's done for all other states except errors.

Reply:
On second thought, might end up having a slightly different fix, as coop close shouldn't transition to `StateContractClosed`

Reply:
oh yeah, good catch